mirascope/core/gemini/_utils/_setup_call.py

```diff
@@ -53,7 +58,8 @@ def setup_call(
     elif extract:
         assert tool_types, "At least one tool must be provided for extraction."
         tool_config = call_kwargs.get("tool_config", {})
-        tool_config["function_calling_config"] = {"mode": "auto"}
+        # Ignore type checking, because gemini tool config is not yet typed correctly
+        tool_config["function_calling_config"] = {"mode": "auto"}  # type: ignore
```

Comment:
I have looked at this, but I think there is a problem with the type definition on the gemini side.
I have not read all of the gemini code due to time constraints, but I hope to find time in the future to go back and fix it.